### PR TITLE
Set title prop in StringCellRenderer

### LINKED ISSUE #223: Set title property in StringCellRenderer
* Release number or git hash: v4.0.0-alpha.12
 * Web browser version and OS: any

### Steps to reproduce

 1. Use longer strings as values in a `StringColumn`.
 1. Hover over them -> no tooltip is shown even though the text is truncated. 

### Observed behavior
The `title` property is missing (or intentionally left out) for `StringCellRenderer`. 

### Expected behavior
Adding the `title` property would allow the browser to automatically show a tooltip on hover.

### PR DESCRIPTION
**Pull request by puehringer**
closes #223 

**prerequisites**: 
 * [x] branch is up-to-date with the branch to be merged with, i.e. develop
 * [x] build is successful
 * [x] code is cleaned up and formatted 
 * [x] tested with Firefox 52, Firefox 57+, Chrome 64+, MS Edge 16+

### Summary
Added `title` property to nodes rendered with `StringCellRenderer` to show a tooltip of the text on hover.

### CODE REVIEW

File: src/renderer/utils.ts [hunk 1]
```diff
@@ -31,6 +31,9 @@ export function setText<T extends Node>(node: T, text?: string): T {
   if (node.textContent !== text) {
     node.textContent = text;
   }
+  if (node instanceof HTMLElement && node.title !== text) {
```
Comment (sgratzl) [inline src/renderer/utils.ts]: I would avoid having a function that is just called `setText` to also automatically to other things. At least from a programmer point of view I wouldn't expect this behavior.

File: src/renderer/StringCellRenderer.ts [hunk 2]
```diff
@@ -24,10 +24,12 @@ export default class StringCellRenderer implements ICellRendererFactory {
...
       update: (n: HTMLDivElement, d: IDataRow) => {
         renderMissingDOM(n, col, d);
+        const label = col.getLabel(d);
         if (col.escape) {
-          setText(n, col.getLabel(d));
+          setText(n, label);
         } else {
-          n.innerHTML = col.getLabel(d);
+          n.innerHTML = label;
+          n.title = label;
```
Comment (sgratzl) [inline src/renderer/StringCellRenderer.ts]: see comment below. 

Moreover, what happens if the text is very long (which might be especially true in the group case) or if the text has HTML elements in it?

Comment (puehringer) [reply src/renderer/StringCellRenderer.ts]: For the HTML elements, one could use `n.title = n.textContent;`, which would return only the text without the tags. 

For very long text, browsers usually truncate the title tooltip, although I agree that large tooltips might annoy users. But on the other hand, what else would be an acceptable solution (for overflowing, but not very long text)? 

I am happy to hear about your input in this matter!

File: src/renderer/utils.ts (hunk 1 above)
Comment (puehringer) [reply src/renderer/utils.ts]: The function is used by several renderers, all of which might benefit from having a `title` property set. Should I refactor the code to only modify the title of `StringCellRenderer` or would it be beneficial to have a (differently named) method setting both text and title?

**Pull request merged**